### PR TITLE
fix: pass response headers from app to router

### LINKED ISSUE #605: add_response_header may fail sometimes
### Bug Description

```python
from robyn import Robyn, jsonify

app = Robyn(__file__)


@app.get("/")
async def h(request):
    return jsonify({"chinese": "中文"})


app.add_response_header("content-type", "application/json; charset=UTF-8")
app.start()
```

when refresh chrome, the output display two results:

`{"chinese":"中文"}`
or
`{"chinese":"涓枃"}`

so, I check it in DevTools,  the `Content-Type` change sometimes

`application/json; charset=UTF-8`
or
`text/plain`

### Steps to Reproduce

_No response_

### Your operating system

Windows

### Your Python version (`python --version`)

3.8

### Your Robyn version

None

### Additional Info

INFO:robyn.logger:Robyn version: 0.37.0

Python 3.8.3

### PR DESCRIPTION
**Pull request by IdoKendo**
**Description**

This PR fixes #605

There was a conflict of Content-Type between what is defined in `app.response_headers` and having it being overridden in the `Router` class. I am unsure why this caused them to sometimes return `text/plain` and sometimes return the value defined in `app.response_headers`, but this fix guarantees that `app.response_headers` is the header that will be set in the `Response`.

### CODE REVIEW

File: robyn/router.py [hunk 1]
```diff
@@ -79,25 +89,42 @@ def add_route(
         handler: Callable,
         is_const: bool,
         exception_handler: Optional[Callable],
+        default_response_headers: List[Header],
     ) -> Union[Callable, CoroutineType]:
+        response_headers = {
+            d[0]: d[1] for d in (h.as_list() for h in default_response_headers)
+        }
```
Comment (sansyrox) [inline robyn/router.py:96]: Hey @IdoKendo 👋 

Great work as usual 😄  However, can we do something else instead of nested interpolation?

Comment (IdoKendo) [reply robyn/router.py:96]: Hey @sansyrox 👋 
Haha, yeah I was also not very satisfied with this line, but honestly the `Header` API is kinda awkward, so it's a bit tricky to do it any other way without introducing tons of boilerplate. I can try to fiddle with it so it's a bit nicer, or we can open a separate issue for it. WDYT?

Comment (sansyrox) [reply robyn/router.py:96]: @IdoKendo , let us just use two for loops for now. 

> I can try to fiddle with it so it's a bit nicer, or we can open a separate issue for it. WDYT?

I think we should definitely investigate 😄 But I will not consider it a blocker for this PR

File: robyn/router.py [hunk 2]
```diff
@@ -79,25 +89,41 @@ def add_route(
         handler: Callable,
         is_const: bool,
         exception_handler: Optional[Callable],
+        default_response_headers: List[Header],
     ) -> Union[Callable, CoroutineType]:
+        headers_list = (h.as_list() for h in default_response_headers)
+        response_headers = {d[0]: d[1] for d in headers_list}
```
Comment (sansyrox) [inline robyn/router.py:95]:
```suggestion
        response_headers = {h.key: h.val for h in default_response_headers}
```

Hey @IdoKendo 👋 

Can we do this instead?

**Review by sansyrox — APPROVED** [review]
LGTM! 🔥 Great work @IdoKendo ✨

**Pull request merged**